draft-ietf-taps-arch.md

```diff
@@ -186,55 +188,63 @@ The Transport Services architecture is based on the survey of Services Provided
 
-Functionality that is common across multiple transport protocols should be accessible through a unified set of API calls. An application should be able to implement logic for its basic use of transport networking (establishing the transport, and sending and receiving data) once, and expect that implementation to continue to function as the transports change.
+Functionality that is common across multiple transport protocols SHOULD be accessible through a unified set of API calls. An application ought to be able to implement logic for its basic use of transport networking (establishing the transport, and sending and receiving data) once, and expect that implementation to continue to function as the transports change.
 
-Any Transport Services API must allow access to the distilled minimal set of features offered by transport protocols {{I-D.ietf-taps-minset}}.
+Any Transport Services API is REQUIRED to allow access to the distilled minimal set of features offered by transport protocols {{I-D.ietf-taps-minset}}.
 
 ## Access to Specialized Features
 
-There are applications that will need to control fine-grained details of transport protocols to optimize their behavior and ensure compatibility with remote peers,. A Transport Services system will therefore also needs to allow more specialized protocol features to be used. The interface for these specialized options should be exposed differently from the common options to ensure flexibility.
+There are applications that will need to control fine-grained details of transport protocols to optimize their behavior and ensure compatibility with remote systems. A Transport Services system therefore SHOULD also permit more specialized protocol features to be used. The interface for these specialized options should be exposed differently from the common options to ensure flexibility.
 
-A specialized feature could be required by an application only when using a specific protocol, and not when using others. For example, if an application is using UDP, it could require control over the checksum or fragmentation behavior for UDP; if it used a protocol to frame its data over a byte stream like TCP, it would not need these options. In such cases, the API should expose the features in such a way that they take effect when a particular protocol is selected, but do not imply that only that protocol could be used if there are equivalent options.
+A specialized feature could be required by an application only when using a specific protocol, and not when using others. For example, if an application is using UDP, it could require control over the checksum or fragmentation behavior for UDP; if it used a protocol to frame its data over a byte stream like TCP, it would not need these options. In such cases, the API ought to expose the features in such a way that they take effect when a particular protocol is selected, but do not imply that only that protocol could be used if there are equivalent options.
```

Comment:
"if there are equivalent options" sounds a bit more limiting / stricter than I think it should. Just as a suggestion, why not say it like this: "... that only that protocol could be used. For example, allowing an application to specify constrained checksum usage will not make communication fail when using a protocol such as TCP, which always uses a checksum covering the entire payload."

Reply:
The approach is OK, is this better:
"For example, if an application were to specify constrained checksum usage, communication would not fail when a protocol such as TCP is selected, which always uses a checksum covering the entire payload.""

Reply:
Great, fixed!